### DISCUSSION TITLE
Do we have support for rtc subsecond calculation in zephyr.

### DISCUSSION THREAD

**Discussion by VinayShirol-eaton** [zephyrproject-rtos/zephyr#53428] [Q&A]
Hi,
I am working with stm32_U5 series board. This board has internal rtc and it also have the subsecond support.

Does zephyr provide internal rtc support?
If yes, How can I achieve the subsecond support through the zephyr api? Could you please suggest me some example/sample code for this.

**Comment by VinayShirol-eaton** [comment]
Hi ,

I was able to read the subsecond register by adding this line of code , [uint32_t subsecond = READ_REG(RTC->SSR);]
It has 3.90625ms resolution [i.e 1/255]. Since the rtc SSR reg is relaoded with PREDIV_S value on each second in BCD mode. The PREDIV_S default value is 0xff.

Do we have any api's in zephyr to read the subsecond register?

Thanks

**Comment by yashi** [comment]
> Does zephyr provide internal rtc support?

Not right now.  There is #35333, #23526, and #52618 but not merged.
See also https://github.com/zephyrproject-rtos/zephyr/projects/18?card_filter_query=rtc
